pkg/hubble/peer/handler.go

```diff
@@ -143,3 +152,30 @@ func nodeAddress(n types.Node) string {
 	}
 	return addr.String()
 }
+
+// tlsServerName constructs a server name to be used as the TLS server name.
+// The server name is of the following form:
+//
+//     <nodeName>.<clusterName>.<hubble-grpc-svc-name>.<domain>
+//
+// For example, with nodeName=moseisley and clusterName=tatooine, the following
+// server name is returned:
+//
+//     moseisley.tatooine.hubble-grpc.cilium.io
+//
+// If nodeName is not provided, an empty string is returned. If clusterName is
+// not provided, it defaults to the default cluster name.
+func tlsServerName(nodeName, clusterName string) string {
+	if nodeName == "" {
+		return ""
```

Comment:
why without nodename does it default to empty string rather than the cluster name?

Reply:
Without `nodename`, it's not possible to construct a valid server name that can be used to match against the TLS certificate.
Thus, in this case, I think it's better to return an empty string rather than an invalid server name.